zebra-state/src/sled_state/sled_format.rs

```diff
@@ -243,7 +243,10 @@ impl SledSerialize for sled::transaction::TransactionalTree {
     {
         let key_bytes = key.into_ivec();
         let value_bytes = value.into_ivec();
-        self.insert(key_bytes, value_bytes)?;
+        let previous = self.insert(key_bytes, value_bytes)?;
+
+        assert!(previous.is_none());
```

Comment:
Can we add a message to the assert, so that users have some context for the failure?

```suggestion
        assert!(previous.is_none(), format!("state insertion error: duplicate key {:?} when inserting value {:?}", key, value));
```